src/main.rs

```diff
@@ -41,8 +41,13 @@ async fn main() {
     let addr = format!("{}:{}", host, port);
     let mut listener = listen(addr, root).await.unwrap();
 
-    if open {
-        let link = listener.link().unwrap();
+    if open.is_some() {
+        let path = if let Some(page) = open.unwrap() {
+            format!("/{page}")
+        } else {
+            String::from("")
+        };
+        let link = listener.link().unwrap() + &path;
```

Comment:
The code could be more simplified:
```rust
if let Some(page) = open {
    let origin = listener.link().unwrap();
    let path = page.unwrap_or_default();
    open::that(format!("{origin}/{path}")).unwrap();
}
```

Reply:
Oh nice. I missed `unwrap_or_default()`, which is useful here. Pushing an update.